homeassistant/components/sensor/filter.py

```diff
@@ -195,6 +225,31 @@ def device_state_attributes(self):
         return state_attr
 
 
+class FilterState(object):
+    """State abstraction for filter usage."""
+
+    def __init__(self, state):
+        """Initialize with HA State object."""
+        self.timestamp = state.last_updated
```

Comment:
Why call it `timestamp` instead of keeping the name `last_updated`? That will cause less confusion.

Reply:
When working with time series, the name of the dimension is always timestamp. This is the correct name in the scientific area.

What causes confusion is last_updated vs last_changed. With the FilterState class I intended to isolate filter algorithms as much as possible from the guts of HA core.